cmd/commands.go

```diff
@@ -0,0 +1,190 @@
+package xcaddycmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/caddyserver/xcaddy"
+	"github.com/caddyserver/xcaddy/internal/utils"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	buildCommand.Flags().StringArray("with", []string{}, "can be used multiple times to add plugins by specifying the Go module name and optionally its version, similar to go get. Module name is required, but specific version and/or local replacement are optional.")
+	buildCommand.Flags().String("output", "", "changes the output file.")
+	buildCommand.Flags().StringArray("replace", []string{}, "is like --with, but does not add a blank import to the code; it only writes a replace directive to go.mod, which is useful when developing on Caddy's dependencies (ones that are not Caddy modules). Try this if you got an error when using --with, like \"cannot find module providing package\".")
+	buildCommand.Flags().StringArray("embed", []string{}, "can be used multiple times to embed directories into the built Caddy executable. The directory can be prefixed with a custom alias and a colon : to use it with the root directive and sub-directive.")
+}
+
+var versionCommand = &cobra.Command{
+	Use:   "version",
+	Long:  "Command to get xcaddy version",
+	Short: "xcaddy version",
+	RunE: func(cm *cobra.Command, args []string) error {
+		fmt.Println(xcaddyVersion())
+		return nil
+	},
+}
+
+var buildCommand = &cobra.Command{
+	Use: `build [<caddy_version>]
+    [--output <file>]
+    [--with <module[@version][=replacement]>...]
+    [--replace <module[@version]=replacement>...]
+    [--embed <[alias]:path/to/dir>...]`,
+	Long: `
+<caddy_version> is the core Caddy version to build; defaults to CADDY_VERSION env variable or latest.
+This can be the keyword latest, which will use the latest stable tag, or any git ref such as:
+
+A tag like v2.0.1
+A branch like master
+A commit like a58f240d3ecbb59285303746406cab50217f8d24
+
+Flags: 
+ --output changes the output file.
+
+ --with can be used multiple times to add plugins by specifying the Go module name and optionally its version, similar to go get. Module name is required, but specific version and/or local replacement are optional.
+
+ --replace is like --with, but does not add a blank import to the code; it only writes a replace directive to go.mod, which is useful when developing on Caddy's dependencies (ones that are not Caddy modules). Try this if you got an error when using --with, like cannot find module providing package.
+
+ --embed can be used multiple times to embed directories into the built Caddy executable. The directory can be prefixed with a custom alias and a colon : to use it with the root directive and sub-directive.
```

Comment:
Do you think the alias thing could use an example?

Reply:
Added an example for the alias.